src/funcx_common/messagepack/exceptions.py

```diff
@@ -11,3 +11,10 @@ class UnrecognizedProtocolVersion(InvalidMessageError):
     When attempting to unpack a message, found a protocol version which was not
     supported.
     """
+
+
+class WrongMessageTypeError(ValueError):
```

Comment:
IMO this should subclass `InvalidMessageError` since having an incorrect message type means the message as a whole is invalid (and since `InvalidMessageError` is a `ValueError` as well)

Reply:
I thought about it both ways and came out with no clear opinion. Since you have a preference, I'll update it as such.
I need to add a changelog anyway to make the build pass. (Whoops.)